Add 'clean' target to ui Makefile

cuongdo opened the pull request:
I was running into issues where an interrupted make on the UI left behind a corrupted mess of generated / npm installed files in my ui directory. This 'clean' target fixes that kind of situation.

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.svg" height="40" alt="Review on Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/4038)

<!-- Reviewable:end -->

Commit:
Added a 'clean' target to ui/Makefile to wipe all files that aren't
committed to the repo. This can help get the UI directory back to
a known good state.

tamird commented on Makefile:
```diff
@@ -197,6 +197,7 @@ clean:
 	$(GO) clean $(GOFLAGS) -i github.com/cockroachdb/...
 	find . -name '*.test' -type f -exec rm -f {} \;
 	rm -f .bootstrap
+	(cd ui && make clean)
```
`make -C ui clean`

tamird commented on ui/Makefile:
```diff
@@ -143,3 +143,8 @@ $(GOBINDATA_DEBUG_TARGET): $(GOBINDATA_DEBUG_DEPS) bower.installed debug/$(INDEX
+
+.PHONY:
+clean:
+	rm -f npm.installed bower.installed tsd.installed $(INDEX)
```
`git clean -dxf`?

bdarnell replied:
I prefer the other form; I think it's generally better for `make clean` to make an effort to restrict its actions to things that may have been put there by the `Makefile` instead of relying on external metadata and deleting anything unrecognized.

cuongdo commented:
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[Makefile, line 200 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4038#-K9ED3J6WODCnA-XYLr0-r1-200:-K9EJQFnd5LNtT_mxe0E:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/3e74f62673386e234bc8c604e6f2b4f09bdc7281/Makefile#L200)):*
Done.

---

*[ui/Makefile, line 149 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4038#-K9ED3J6WODCnA-XYLr1-r1-149:-K9EJQs2u900McLwL4YJ:794403532) ([raw file](https://github.com/cockroachdb/cockroach/blob/3e74f62673386e234bc8c604e6f2b4f09bdc7281/ui/Makefile#L149)):*
Neat trick. I changed it to `git clean -dXf`. The `-x` is way too dangerous. If someone were working on new .ts file and hadn't done a `git add` on the file, make clean would wipe out their work.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4038)*
<!-- Sent from Reviewable.io -->

tamird commented:
plz squash. LGTM

---

Reviewed 2 of 2 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[ui/Makefile, line 149 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4038#-K9ED3J6WODCnA-XYLr1-r1-149:-K9EJxiYptdqeqI5MPb-:1183338489) ([raw file](https://github.com/cockroachdb/cockroach/blob/3e74f62673386e234bc8c604e6f2b4f09bdc7281/ui/Makefile#L149)):*
Ah, didn't know about `-X`. Neat.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4038#-:-K9EK-KHIw4Lw5cPP6tl:139452876)*
<!-- Sent from Reviewable.io -->

bdarnell commented on ui/Makefile:
```diff
@@ -143,3 +143,7 @@ $(GOBINDATA_DEBUG_TARGET): $(GOBINDATA_DEBUG_DEPS) bower.installed debug/$(INDEX
 watch: debug
 	gulp watch
+
+.PHONY:
```
I don't think `.PHONY:` on its own does anything. It should be `.PHONY: clean` (see the top-level makefile)

cuongdo commented:
Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[ui/Makefile, line 149 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4038#-K9ED3J6WODCnA-XYLr1-r1-149:-K9ERgybJve_WZuSwMtE:-1098504442) ([raw file](https://github.com/cockroachdb/cockroach/blob/3e74f62673386e234bc8c604e6f2b4f09bdc7281/ui/Makefile#L149)):*
OK, changed back.

---

*[ui/Makefile, line 147 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4038#-K9ED3J6WODCnA-XYLr1-r2-147:-K9ERcuYanKDeWVTluMY:1950624466) ([raw file](https://github.com/cockroachdb/cockroach/blob/39e25ca5e2328e2759200de1c19f2def76725f58/ui/Makefile#L147)):*
Fixed that and the other .PHONY declarations in this Makefile.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4038)*
<!-- Sent from Reviewable.io -->

maxlang commented:
LGTM
